pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -2316,6 +2316,7 @@ type SSSDSidecarAdditionalFile struct {
 
 // NetworkSpec for Ceph includes backward compatibility code
 // +kubebuilder:validation:XValidation:message="at least one network selector must be specified when using multus",rule="!has(self.provider) || (self.provider != 'multus' || (self.provider == 'multus' && size(self.selectors) > 0))"
+// +kubebuilder:validation:XValidation:message="the legacy hostNetwork setting can only be set if the network.provider is set to the empty string",rule="(self.hostNetwork != true) || (self.provider  ==  ”)"
```

Comment:
```suggestion
// +kubebuilder:validation:XValidation:message="the legacy hostNetwork setting can only be set if the network.provider is set to the empty string",rule="(self.hostNetwork != true) || (self.provider  ==  '')"
```

Reply:
@subhamkrai  this suggestion essentially undoes my latest update  which was to fix the build/go vet failure. 
The proposed  change is the inverse of the diff that go vet had produced in the CI and locally ...

Reply:
in the meantime, I implemented your suggestion and undid it again since `gofmt` really insists on the strange quote. Additionaly, I added `has() checks to the rule.

Now it  builds just fine locally for me, and I can `kubectl create` the resulting `ceds.yaml` but the created test-cluster from `cluster-test.yaml` still fails to be admitted:

```
Events:
  Type     Reason           Age               From                          Message
  ----     ------           ----              ----                          -------
  Warning  ReconcileFailed  1s (x10 over 4s)  rook-ceph-cluster-controller  failed to reconcile CephCluster "rook-ceph/my-cluster". failed to add finalizer: failed to add finalizer "cephcluster.ceph.rook.io" on "my-cluster": CephCluster.ceph.rook.io "my-cluster" is invalid: spec.network: Invalid value: "object": the legacy hostNetwork setting can only be set if the network.provider is set to the empty string
```

Reply:
The CI also shows that gofmt wants the strange quote: https://github.com/rook/rook/actions/runs/7845731244/job/21410846530?pr=13693

Reply:
It seems like we just need to figure out how to resolve this quoting issue. I can surmise that `''` must not be the correct choice for the empty string. 

When I look at the Common Expression Language (the expression syntax used for VAPs) docs, there is a clear definition of what the empty string, as well as how to escape characters. https://github.com/google/cel-spec/blob/master/doc/langdef.md#string-and-bytes-values

I would advise that escape characters be avoided if possible. To this end, Kubebuilder allows using the same golang string syntaxes we are used to, and [reportedly](https://github.com/kubernetes-sigs/controller-tools/issues/315#issuecomment-532407196) we can use the grave accent (or "backtick")(\`) to define the kubebuilder strings.

I'd suggest seeing if the grave accents help, like this:
```
// +kubebuilder:validation:XValidation:message=`msg`,rule=`thing == ""`
```

Reply:
@BlaineEXE  I just tried the backticks quoting locally and `make crds` ran fine but produced no diff on the crds. 

Soi I conclude  that the quoting is not the problem with this PR